sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletions/ChatMessageImageUrl.Serialization.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI;
+
+[CodeGenSuppress("global::System.ClientModel.Primitives.IJsonModel<Azure.AI.OpenAI.ChatMessageImageUrl>.Write", typeof(Utf8JsonWriter), typeof(ModelReaderWriterOptions))]
+[CodeGenSuppress("DeserializeChatMessageImageUrl", typeof(JsonElement), typeof(ModelReaderWriterOptions))]
+public partial class ChatMessageImageUrl : IUtf8JsonSerializable, IJsonModel<ChatMessageImageUrl>
+{
+    void IJsonModel<ChatMessageImageUrl>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
```

Comment:
Instead of overriding the entire serialization method can we instead override just the properties that need to be changed?  https://github.com/Azure/autorest.csharp/blob/feature/v3/readme.md#customize-serializationdeserialization-methods the goal is to minimize the custom code footprint where possible.